proto/viam/app/v1/app.proto

```diff
@@ -1162,6 +1165,15 @@ message DeleteKeyRequest {
 
 message DeleteKeyResponse {}
 
+message RenameKeyRequest {
+  string id = 1;
+  string new_name = 2;
+}
+
+message RenameKeyResponse {
+  string token_name = 1;
```

Comment:
what is this? I would expect an empty response.

Reply:
you can blame me! i told him above to parrot it back in case we did any stripping of special characters etc and the name was slightly different than whatever was request. if you hate it though, i dont feel strongly so fine to remove.

Reply:
I agreed that it was a good sanity check. Though I didn't think about whether it would conform to the expected behavior of the other related rpcs.

Reply:
Do our other Key api calls change the key "name" if it is invalid?

Reply:
Whoops, will try to un-stall this PR. @mattjperez, please ignore this noise and focus on your new team! :) 

@stevebriskin:
`CreateKeyResponse` [above](https://github.com/viamrobotics/api/pull/489/files/b758ad01362595f998d4f1d9a416d47e2db24bb8#diff-e842ca0f50219c67d40be90a2a1efd40ee1b6df19a254357ab177d631b3ea4b4R1157-R1160):
```
message CreateKeyResponse {
  string key = 1;
  string id = 2;
}
```
i.e. we echo back the provided name and send the newly generated ID.

What if I updated these new RPCs to match a bit better:
```
message RenameKeyRequest {
  string id = 1;
  string name = 2;
}

message RenameKeyResponse {
  string name = 1;
}
```

Reply:
👍

Reply:
Changing `RenameKeyResponse` to be:
```
message RenameKeyResponse {
  string id = 1;
  string name = 2;
}
```
after in person conversation